src/lib/Tests/Pagination/Pagerfanta/URLWildcardAdapterTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Tests\Pagination\Pagerfanta;
+
+use eZ\Publish\API\Repository\URLWildcardService;
+use eZ\Publish\API\Repository\Values\Content\UrlWildcard;
+use Ibexa\AdminUi\Pagination\Pagerfanta\URLWildcardAdapter;
+use PHPUnit\Framework\TestCase;
+
+class URLWildcardAdapterTest extends TestCase
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\UrlWildcard|\PHPUnit\Framework\MockObject\MockObject */
+    private $urlWildcardService;
+
+    protected function setUp(): void
+    {
+        $this->urlWildcardService = $this->createMock(URLWildcardService::class);
+    }
+
+    public function testGetNbResults()
+    {
+        $countAll = 5;
+
+        $this->urlWildcardService
+            ->expects($this->once())
+            ->method('countAll')
+            ->with()
+            ->willReturn($countAll);
+
+        $adapter = new URLWildcardAdapter($this->urlWildcardService);
+
+        $this->assertEquals(
+            $countAll,
+            $adapter->getNbResults()
```

Comment:
We could also assert that calling `getNbResults` does **NOT** load the results, i.e. `urlWildcardService::loadAll` method is not called. The reverse is true for following tests: when loading a result slice, we expect method `urlWildcardService::countAll()` to not be called as well.

Additionally, subsequent calls to `$adapter->getNbResults()` should not call the `urlWildcardService::countAll` method, as seen in your implementation where you store the result in property.
(just call the `$adapter->getNbResults()` a second time here, which will trigger the `self::once()` method call counter assertion if mock is called a second time).